map_conversion.py

```diff
@@ -0,0 +1,250 @@
+                node = ET.SubElement(root, "node")
+                node.set("id", str(global_id))
+                present_nodes[currNode] = global_id
+                global_id += 1
+                node.set("x", str(pose[0]))
+                node.set("y", str(pose[1]))
+            lane_dict[way["token"]].append(present_nodes[currNode])
+    return lane_dict
+
+
+def populate_polys(data: Dict[str, Iterable[Any]]) -> Dict[str, np.ndarray]:
+    """
+    Returns a dictionary poly_dict
+    poly_dict maps a lane polygon in the original map to its exterior points.
+    """
+    # Map nodes in the original json to their x, y coordinates
+    node_dict = {}  # k: node_token v: (x,y)
+
+    # Loop over all nodes in the json to populate the node_dict
+    for node in data["node"]:
+        node_dict[node["token"]] = (node["x"], node["y"])
+
+    # Map polygons in the original json to the x,y coordinates of their exterior nodes
+    poly_dict = {}  # k: poly_token v: np.ndarray of shape (N,2)
+
+    # Loop over all nodes in the json to populate the node_dict
+    for poly in data["polygon"]:
+        poly_array = []
+        for node_id in poly["exterior_node_tokens"]:
+            poly_array.append(list(node_dict[node_id]))
+        poly_dict[poly["token"]] = np.array(poly_array)
+    return poly_dict
+
+
+def create_lanes_xml(
+    nusc_map: NuScenesMap,
+    root: ET.Element,
+    data: Dict[str, Iterable[Any]],
+    filename: str,
+    argo_dir: str,
+    lane_dict: Dict[str, Iterable[int]],
+    poly_dict: Dict[str, np.ndarray],
+) -> None:
+    """
+    Fill up the xml map file with lane centelines.
+    Also create the supporting files halluc_bbox_table.npy and tableidx_to_laneid_map.json
+    """
+    # Id to assign to lanes in the new xml map file. We arbitrarily start with 8000000.
+    # We make up new lane_ids since the original one's are non numerical
+    global_way_id = 8000000
+
+    # Map that links new lane_id in the xml to its original token in the json.
+    way_to_lane_id = {}
+
+    # map lane segment IDs to their index in the table
+    tableidx_to_laneid_map = {}
+    # array that holds xmin,ymin,xmax,ymax for each coord
+    halluc_bbox_table = []
+    table_idx_counter = 0
+
+    ## Iterate over the lanes to create the required xml and supporting files
+    for way in data["lane"]:
+        if way["token"] == "8f23d3ed-1089-4fcf-a178-a174abc69938":
+            continue
+        node = ET.SubElement(root, "way")
+
+        if way["token"] not in way_to_lane_id:
+            way_to_lane_id[way["token"]] = global_way_id
+            global_way_id += 1
+        curr_id = way_to_lane_id[way["token"]]
+
+        node.set("lane_id", str(curr_id))
+        traffic = ET.SubElement(node, "tag")
+        traffic.set("k", "has_traffic_control")
+        traffic.set("v", "False")
+
+        turn = ET.SubElement(node, "tag")
+        turn.set("k", "turn_direction")
+        turn.set("v", "NONE")
```

Comment:
are we setting `turn_direction = None` for all lane segments? I'm guessing nuScenes might not provide that info, so I would make those defaults global variables at the top of the file, then use the default value here:
```python
DEFAULT_TURN_DIRECTION = "NONE"
...
        turn = ET.SubElement(node, "tag")
        turn.set("k", "turn_direction")
        turn.set("v", DEFAULT_TURN_DIRECTION)
```

Reply:
Good idea, done

Reply:
thanks